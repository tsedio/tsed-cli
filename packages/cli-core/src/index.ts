import "./utils/patchCommander.js";

import Inquirer from "inquirer";

export * from "./CliCore.js";
export * from "./decorators/index.js";
export * from "./interfaces/index.js";
export * from "./packageManagers/index.js";
export * from "./services/index.js";
export * from "./utils/index.js";
export * from "@tsed/core";
export * from "@tsed/di";
export * from "@tsed/logger";
export * from "@tsed/normalize-path";
export {Inquirer};
