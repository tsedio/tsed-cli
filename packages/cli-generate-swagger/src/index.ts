export * from "./commands/GenerateSwaggerCmd";

export default {};
