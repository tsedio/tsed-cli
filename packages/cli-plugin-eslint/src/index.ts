import "./templates/eslint.template.js";

import {CliPluginEslintModule} from "./CliPluginEslintModule.js";
export * from "./utils/templateDir.js";

export default CliPluginEslintModule;
