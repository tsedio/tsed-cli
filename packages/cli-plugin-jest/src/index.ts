import {CliPluginJestModule} from "./CliPluginJestModule.js";

export * from "./templates/index.js";

export default CliPluginJestModule;
