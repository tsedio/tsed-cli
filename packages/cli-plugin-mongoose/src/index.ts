import {CliPluginMongooseModule} from "./CliPluginMongooseModule.js";

export * from "./hooks/MongooseGenerateHook.js";
export * from "./hooks/MongooseInitHook.js";
export * from "./services/CliMongoose.js";

export default CliPluginMongooseModule;
