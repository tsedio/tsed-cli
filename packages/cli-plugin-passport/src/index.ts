import "./templates/index.js";

import {CliPluginPassportModule} from "./CliPluginPassportModule.js";

export * from "./services/PassportClient.js";

export default CliPluginPassportModule;
