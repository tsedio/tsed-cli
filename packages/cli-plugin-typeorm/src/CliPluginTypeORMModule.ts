import type {AlterPackageJson, RenderDataContext} from "@tsed/cli";
import {inject, ProjectPackageJson} from "@tsed/cli-core";
import {injectable} from "@tsed/di";

import {TypeORMGenerateHook} from "./hooks/TypeORMGenerateHook.js";
import {TypeORMInitHook} from "./hooks/TypeORMInitHook.js";

export class CliPluginTypeORMModule implements AlterPackageJson {
  protected packageJson = inject(ProjectPackageJson);

  $alterPackageJson(packageJson: ProjectPackageJson, data: RenderDataContext) {
    if (data.typeorm) {
      packageJson.addDependencies({
        typeorm: "latest"
      });
    }

    return packageJson;
  }
}

injectable(CliPluginTypeORMModule).imports([TypeORMInitHook, TypeORMGenerateHook]);
