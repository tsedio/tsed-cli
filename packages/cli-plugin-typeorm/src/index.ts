import "./templates/datasource.template.js";

import {CliPluginTypeORMModule} from "./CliPluginTypeORMModule.js";

export * from "./hooks/TypeORMGenerateHook.js";
export * from "./hooks/TypeORMInitHook.js";

export default CliPluginTypeORMModule;
