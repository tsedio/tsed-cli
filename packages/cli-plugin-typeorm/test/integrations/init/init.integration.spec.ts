import {PackageManager} from "@tsed/cli-core";
import {CliPlatformTest, FakeCliFs} from "@tsed/cli-testing";
import {ArchitectureConvention, FeatureType, InitCmd, PlatformType, ProjectConvention, TEMPLATE_DIR} from "@tsed/cli";
import "../../../src";
import "@tsed/cli-plugin-jest";

describe("TypeORM: Init cmd", () => {
  beforeEach(() => {
    return CliPlatformTest.bootstrap({
      templateDir: TEMPLATE_DIR,
      commands: [InitCmd],
      argv: ["init"]
    });
  });
  afterEach(() => CliPlatformTest.reset());

  it("should generate a project with the right options", async () => {
    CliPlatformTest.setPackageJson({
      name: "",
      version: "1.0.0",
      description: "",
      scripts: {},
      dependencies: {},
      devDependencies: {}
    });

    await CliPlatformTest.exec("init", {
      verbose: false,
      root: ".",
      tsedVersion: "5.58.1",
      rootDir: "./.tmp/init/default",
      production: false,
      rawArgs: [],
      platform: PlatformType.EXPRESS,
      architecture: ArchitectureConvention.DEFAULT,
      convention: ProjectConvention.DEFAULT,
      packageManager: PackageManager.YARN,
      projectName: "default",
      db: true,
      typeorm: true,
      mysql: true,
      features: [FeatureType.DB, FeatureType.TYPEORM, FeatureType.TYPEORM_MYSQL],
      srcDir: "src",
      pnpm: false,
      npm: false,
      yarn: true,
      express: true,
      koa: false,
      platformSymbol: "PlatformExpress"
    });

    expect(FakeCliFs.getKeys()).toMatchInlineSnapshot(`
      Array [
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/jest.config.js",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.integration.spec.ts",
        "project-name/src/Server.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.integration.spec.ts",
        "project-name/src/controllers/rest/HelloWorldController.spec.ts",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/datasources",
        "project-name/src/datasources/MysqlDatasource.spec.ts",
        "project-name/src/datasources/MysqlDatasource.ts",
        "project-name/src/index.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json",
      ]
    `);

    const content = FakeCliFs.entries.get("project-name/src/Server.ts")!;
    expect(content).toMatchSnapshot();

    const datasource = FakeCliFs.entries.get("project-name/src/datasources/MysqlDatasource.ts")!;
    expect(datasource).toMatchSnapshot();

    const spec = FakeCliFs.entries.get("project-name/src/datasources/MysqlDatasource.spec.ts")!;
    expect(spec).toMatchSnapshot();
  });
  it("should not generate database if any option is selected", async () => {
    CliPlatformTest.setPackageJson({
      name: "",
      version: "1.0.0",
      description: "",
      scripts: {},
      dependencies: {},
      devDependencies: {}
    });

    await CliPlatformTest.exec("init", {
      verbose: false,
      root: ".",
      tsedVersion: "5.58.1",
      rootDir: "./.tmp/init/default",
      production: false,
      rawArgs: [],
      platform: "express",
      architecture: "arc_default",
      convention: "conv_default",
      packageManager: "yarn",
      projectName: "default",
      db: true,
      typeorm: true,
      mysql: true,
      features: [],
      srcDir: "src",
      pnpm: false,
      npm: false,
      yarn: true,
      express: true,
      koa: false,
      platformSymbol: "PlatformExpress"
    });

    expect(FakeCliFs.getKeys()).toMatchInlineSnapshot(`
      Array [
        "./project-name",
        "project-name",
        "project-name/.barrelsby.json",
        "project-name/.dockerignore",
        "project-name/.gitignore",
        "project-name/Dockerfile",
        "project-name/README.md",
        "project-name/docker-compose.yml",
        "project-name/jest.config.js",
        "project-name/package.json",
        "project-name/processes.config.js",
        "project-name/src",
        "project-name/src/Server.integration.spec.ts",
        "project-name/src/Server.ts",
        "project-name/src/config",
        "project-name/src/config/envs",
        "project-name/src/config/envs/index.ts",
        "project-name/src/config/index.ts",
        "project-name/src/config/logger",
        "project-name/src/config/logger/index.ts",
        "project-name/src/controllers/rest",
        "project-name/src/controllers/rest/HelloWorldController.integration.spec.ts",
        "project-name/src/controllers/rest/HelloWorldController.spec.ts",
        "project-name/src/controllers/rest/HelloWorldController.ts",
        "project-name/src/index.ts",
        "project-name/tsconfig.compile.json",
        "project-name/tsconfig.json",
      ]
    `);
  });
});
