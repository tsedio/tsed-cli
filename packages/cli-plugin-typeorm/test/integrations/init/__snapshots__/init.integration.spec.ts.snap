// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypeORM: Init cmd should generate a project with the right options 2`] = `
"import {join} from \\"path\\";
import {Configuration, Inject} from \\"@tsed/di\\";
import {PlatformApplication} from \\"@tsed/common\\";
import \\"@tsed/platform-express\\"; // /!\\\\ keep this import
import \\"@tsed/ajv\\";
import {config} from \\"./config/index\\";
import * as rest from \\"./controllers/rest/index\\";

@Configuration({
  ...config,
  acceptMimes: [\\"application/json\\"],
  httpPort: process.env.PORT || 8083,
  httpsPort: false, // CHANGE
  disableComponentsScan: true,
  ajv: {
    returnsCoercedValues: true
  },
  mount: {
    \\"/rest\\": [
      ...Object.values(rest)
    ]
  },
  middlewares: [
    \\"cors\\",
    \\"cookie-parser\\",
    \\"compression\\",
    \\"method-override\\",
    \\"json-parser\\",
    { use: \\"urlencoded-parser\\", options: { extended: true }}
  ],
  views: {
    root: join(process.cwd(), \\"../views\\"),
    extensions: {
      ejs: \\"ejs\\"
    }
  },
  exclude: [
    \\"**/*.spec.ts\\"
  ]
})
export class Server {
  @Inject()
  protected app: PlatformApplication;

  @Configuration()
  protected settings: Configuration;
}
"
`;

exports[`TypeORM: Init cmd should generate a project with the right options 3`] = `
"import {registerProvider} from \\"@tsed/di\\";
import {DataSource} from \\"typeorm\\";
import {Logger} from \\"@tsed/logger\\";

export const MYSQL_DATASOURCE = Symbol.for(\\"MysqlDatasource\\");
export const MysqlDatasource = new DataSource({
  type: \\"mysql\\",
  entities: [],
  host: \\"localhost\\",
  port: 3306,
  username: \\"test\\",
  password: \\"test\\",
  database: \\"test\\"
});

registerProvider<DataSource>({
  provide: MYSQL_DATASOURCE,
  type: \\"typeorm:datasource\\",
  deps: [Logger],
  async useAsyncFactory(logger: Logger) {
    await MysqlDatasource.initialize();

    logger.info(\\"Connected with typeorm to database: Mysql\\");

    return MysqlDatasource;
  },
  hooks: {
    $onDestroy(dataSource) {
      return dataSource.isInitialized && dataSource.close();
    }
  }
});
"
`;

exports[`TypeORM: Init cmd should generate a project with the right options 4`] = `
"import { PlatformTest } from \\"@tsed/common\\";
import { MysqlDatasource } from \\"./MysqlDatasource\\";

describe(\\"MysqlDatasource\\", () => {
  beforeEach(PlatformTest.create);
  afterEach(PlatformTest.reset);

  it(\\"should do something\\", () => {
    const instance = PlatformTest.get<MysqlDatasource>(MysqlDatasource);
    // const instance = PlatformTest.invoke<MysqlDatasource>(MysqlDatasource); // get fresh instance

    expect(instance).toBeInstanceOf(MysqlDatasource);
  });
});
"
`;
